yfinance/base.py

```diff
@@ -205,6 +207,14 @@ def history(self, period="1mo", interval="1d",
             debug_mode = kwargs["debug"]
 
         err_msg = "No data found for this date range, symbol may be delisted"
+
+        if data is None or not type(data) is dict or 'status_code' in data.keys():
+            shared._DFS[self.ticker] = utils.empty_df()
+            shared._ERRORS[self.ticker] = err_msg
+            if "many" not in kwargs and debug_mode:
+                print('- %s: %s' % (self.ticker, err_msg))
+            return utils.empty_df()
+
         if "chart" in data and data["chart"]["error"]:
```

Comment:
NoneType is not iterable

Reply:
The return statement should resolve any issues here.

Reply:
My mistake, I was so focused on what I changed I didn't literally read between the lines 😅